Client/States/GameplayState.cpp

```diff
@@ -49,56 +57,115 @@ void GameplayState::InitCanvas(){
 
 void GameplayState::InitCrossHeir(){
     //crossheir
-    auto crossHeirVert = canvas->AddElement()
-            .SetColor({1.0,1.0,1.0,1.0})
-            .SetAbsoluteSize({15,3})
-            .AlignCenter()
-            .AlignMiddle();
-
-    auto crossHeirHoriz = canvas->AddElement()
-            .SetColor({1.0,1.0,1.0,1.0})
-            .SetAbsoluteSize({3,15})
-            .AlignCenter()
-            .AlignMiddle();
 
+    crosshair = &canvas->AddImageElement(crosshairImage)
+        .SetAbsoluteSize({ 32, 32 })
+        .CenterSprite();
+    crosshair->OnUpdate.connect<&GameplayState::UpdateCrosshair>(this);
 }
 
 void GameplayState::InitTimerBar(){
     //timer Bar
-    timeBar = &canvas->AddElement()
-            .SetColor({0,1,0,1})
-            .SetAbsoluteSize({800, 30})
+    int outlineSize = timerBarOutline;
+    if (baseClient->IsSinglePlayer())
```

Comment:
BaseClient really shouldn't be used after the client thread is created! The is singleplayer check to before the thread creation to ensure thread safety. You don't need to repeatedly call it in game in any case.

Reply:
![istockphoto-182060276-612x612](https://github.com/The-Newcastle-Gang/SpeedFun/assets/122457074/820ae8f9-9e2c-4799-b744-b097b709eda6)